code/__DEFINES/icon_smoothing.dm

```diff
@@ -0,0 +1,124 @@
+/* smoothing_flags */
+/// Smoothing system in where adjacencies are calculated and used to build an image by mounting each corner at runtime.
+#define SMOOTH_CORNERS (1<<0)
+/// Smoothing system in where adjacencies are calculated and used to select a pre-baked icon_state, encoded by bitmasking.
+#define SMOOTH_BITMASK (1<<1)
+/// Atom has diagonal corners, with underlays under them.
+#define SMOOTH_DIAGONAL_CORNERS	(1<<2)
+/// Atom will smooth with the borders of the map.
+#define SMOOTH_BORDER	(1<<3)
+/// Atom is currently queued to smooth.
+#define SMOOTH_QUEUED	(1<<4)
+/// Smooths with objects, and will thus need to scan turfs for contents.
+#define SMOOTH_OBJ		(1<<5)
+
+
+DEFINE_BITFIELD(smoothing_flags, list(
+	"SMOOTH_CORNERS" = SMOOTH_CORNERS,
+	"SMOOTH_BITMASK" = SMOOTH_BITMASK,
+	"SMOOTH_DIAGONAL_CORNERS" = SMOOTH_DIAGONAL_CORNERS,
+	"SMOOTH_BORDER" = SMOOTH_BORDER,
+	"SMOOTH_QUEUED" = SMOOTH_QUEUED,
+	"SMOOTH_OBJ" = SMOOTH_OBJ,
+))
+
+
+/*smoothing macros*/
+
+#define QUEUE_SMOOTH(thing_to_queue) if(thing_to_queue.smoothing_flags & (SMOOTH_CORNERS|SMOOTH_BITMASK)) {SSicon_smooth.add_to_queue(thing_to_queue)}
+
+#define QUEUE_SMOOTH_NEIGHBORS(thing_to_queue) for(var/atom/atom_neighbor as anything in orange(1, thing_to_queue)) {QUEUE_SMOOTH(atom_neighbor)}
```

Comment:
This is pretty horrible, having some kind of 3D array on a subsystem (Or any other spatial data structure) would allow for much better optimisations but out of scope I guess? Having to queue EVERYTHING for smoothing even if its not part of the same smoothing group seems super jank

Reply:
(Actually it was a problem before, so I guess this PR is still an improvement)